flows/interfaces.go

```diff
@@ -244,23 +244,26 @@ type Trigger interface {
 	types.XValue
 	types.XResolvable
 
+	Initialize(Session) error
+	InitializeRun(FlowRun, Step) error
```

Comment:
not the greatest naming but triggers need to do 2 things - setup the session and then additionally do stuff to the first run, i.e. if we have a message, we need to set it as input on the first run